### DISCUSSION TITLE
Can we use value objects in Blazor components?

### DISCUSSION THREAD

**Discussion by MrYossu** [SteveDunn/Vogen#731] [Q&A]
If we have an object with (say) a `string` property, then in a Blazor `EditForm` we can do the following...

```xml
<InputText @bind-Value="MyModel.Name" />
```

However, if `name` is a value object, then this doesn't work, as (presumably) the `InputText` component assumes that the property being bound has a public getter and setter. As value objects don't have those, it looks like we can use them for any properties that we want to edit in a Blazor form.

Is there any way around this? I know I can create a model for the form which uses the underlying primitive data type, but apart from adding yet another representation of the entity (on top of the database model and the DTO), it kind of removes the basic advantage of VOs in the first place.

Anyone able to help here? Thanks

**Comment by SteveDunn** [comment]
I would guess the reason is very similar to why it can't be used in routes: https://github.com/SteveDunn/Vogen/discussions/729

I'll keep an eye on this changing in Blazor and as soon as it's possible, I'll update Vogen (I too would love to use Vogen in forms, routes, and parameters)
